test/test_transformers.py

```diff
@@ -1635,6 +1638,7 @@ def test_invalid_fused_inputs_attn_mask_present(self, device, kernel: SDPBackend
                 q, k, v, mask, 0.0, False))
 
     @onlyCUDA
+    @xfailIfRocm  # ROCm GPU kernels accepts irregular shapes and thus assertRaises will fail
```

Comment:
Wouldn't it be better to change `self.assertRaises(RuntimeError, lambda: ` on line 1650 with
```python
   ctxmgr = self.assertRaises(RuntimeError) if not TEST_WITH_ROCM else contextlib.nullcontext()
   with ctxmgr:
       torch.nn.functional.scaled_dot_product_attention(q, k, v, None, 0.0, False)
```